test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -294,6 +300,67 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 sendMalformedRequest: true);
         }
 
+        [Fact]
+        public async Task OnCompletedExceptionShouldNotPreventAResponse()
+        {
+            var hostBuilder = TransportSelector.GetWebHostBuilder()
+                .UseKestrel()
+                .UseUrls("http://127.0.0.1:0/")
+                .Configure(app =>
+                {
+                    app.Run(async context =>
+                    {
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+                        context.Response.OnCompleted(async () => throw new Exception());
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+                        await context.Response.WriteAsync("hello, world");
+                    });
+                });
+
+            using (var host = hostBuilder.Build())
+            {
+                host.Start();
+
+                using (var client = new HttpClient())
+                {
+                    var response = await client.GetAsync($"http://127.0.0.1:{host.GetPort()}/");
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task OnCompletedShouldNotBlockAResponse()
+        {
+            var delay = Task.Delay(TimeSpan.FromSeconds(20));
```

Comment:
Nit: Use `TestConstants.DefaultTimeout`. If that's too short we might want to consider making the constant larger.

Reply:
Will do